### ISSUE TITLE
New /resources, blog, and podcast view

### ISSUE THREAD

**Issue by Atmosfearful** [KlimaDAO/klimadao#534]
# Summary
As the amount of content we create increases, and the complexity of our product grows, we need a new integrated design that exposes users to all of the various kinds of content and guides that we are creating. Creative team has come up with a nice new home for all of our content. This feature includes:
- A `tag` property for all documents in our CMS that lets us do more advanced querying
- A Sidebar that lets users select which tags to filter by in the UI
- A simple text search

### DESIGN
https://www.figma.com/file/NQTGQDrS7PEfB3QOzYMBN3/RESOURCES?node-id=5%3A833

### PREVIEW LINK

STAGING (merged code status): https://staging-site.klimadao.finance/resources
DEVELOPMENT (unmerged status): https://preview-cms.vercel.app/resources

---

## Todos for GO LIVE :tada:

=> moved to new ticket https://github.com/KlimaDAO/klimadao/issues/760

----

Completion is blocked by #532 because we need to make sure we don't lose access to our /community and /contact pages.

# Implementation Plan
Separate PRs:
* [x] Enhance the CMS to support "tags" property on Podcast and Blog documents => [PR](https://github.com/KlimaDAO/klimadao/pull/609)
- create a new top-level content type called `tag`. A `tag` is a document schema that has these properties:
```ts
interface Tag {
  // slug, not string, because Sanity Slugs can enforce uniqueness https://www.sanity.io/docs/slug-type
  value: Slug;
  // label shown in the nav UI (hardcoded for v1, might be queried from cms in the future)
  label_en: string;
  // just a string for notes, but maybe we can show this in the UI in the far future.
  description_en: string;
}
```
-  [x] for Podcast and Post documents, add a property to the schema: `tags: Reference[]` that is an array of references to tag documents. Sanity Studio users should be able to select from a dropdown of existing tags. See the docs for help on references. => [PR](https://github.com/KlimaDAO/klimadao/pull/609)

https://www.sanity.io/docs/reference-type

---

### SANITY
- [x] Please deploy the new tags to Sanity.io (already merged to Staging)
- [x] Get a list of all Tags the content editor want to use (lists needs to be hard coded in site)
- [x] We need another checkbox for the "FEATURED SLIDER" for Posts, like "is featured article" => [PR](https://github.com/KlimaDAO/klimadao/pull/644)
- [x] We need another Image Upload for the "FEATURED SLIDER" ? **All images in Sanity have text on top !** => NOPE, editors have to ensure to upload a nice image for posts which should appear in the Slider

#### Ticket for Content Editors: 
=> https://github.com/KlimaDAO/klimadao/issues/647

---
- [x] create a new page "resources-2" or alike and HIDE FROM PRODUCTION => [PR](https://github.com/KlimaDAO/klimadao/pull/640)
- [x] Queries: get all documents (podcasts and posts), sorted desc, limit to XY => [PR](https://github.com/KlimaDAO/klimadao/pull/640)
- [x] Queries: get all post with flag "isFeaturedArticle" => [PR](https://github.com/KlimaDAO/klimadao/pull/650)
- [x] Queries: update queries to support "sort by" dropdown, and add it to the ui => [PR](https://github.com/KlimaDAO/klimadao/pull/689)
- [x] Queries: update queries to support "filter by tags" checkboxes => [PR](https://github.com/KlimaDAO/klimadao/pull/689)
- [x] Queries: update queries to support plain text search => [PR](https://github.com/KlimaDAO/klimadao/pull/666)

- [x] UI: Add top section with Slider for Posts => [PR](https://github.com/KlimaDAO/klimadao/pull/650)
- [x] UI: Add Search bar for simple Text search for all Posts => [PR](https://github.com/KlimaDAO/klimadao/pull/666)
- [x] UI: Add SortBy Dropdown which sorts all filtered/non-filtered posts => [PR](https://github.com/KlimaDAO/klimadao/pull/689)
- [x] UI: Add SideNav with hard coded Tags as Checkboxes => [PR](https://github.com/KlimaDAO/klimadao/pull/689)
- [x] ~UI: Add "Share" functionality (How should this look like? Share to where what? Should a Modal open? Examples?)~ => new ticket https://github.com/KlimaDAO/klimadao/issues/749

### Further Notes

- Tags are hard-coded for v1, we consider adding cms control to this as well in a later version if needed.
- user can click an item label in the nav, and it will be selected as the only tag.
- user can click the radio checkbox to add or remove items to the filter
- v1 - /podcast and /blog will redirect to /resources
```
// pseudocode query
get all posts & podcasts where tag includes {tags}
```

# Considerations

### TAGS
What are the TAGs exactly?
We need a list of all tag names from Sanity.

### OPEN UI Questions

- [ ] how many documents should be fetched on initial render? 
- [ ] Do we need a pagination for the list of posts and podcasts?
- [ ] support Podcast in the Featured Slider too ?
- [ ] Add pagination ?

## Redirects

During development:
- Keep redirect "resources" -> "blog" on Production
- Remove this redirect on Staging so that "resources" can be reached

GO LIVE:
- As soon as "resources" can go live, switch the Production redirect to "blog" -> "resources"

**Comment by Atmosfearful** [comment]
> Can you give me feedback in the ticket on how the filtering by tags and sorting by date should happen?

get all documents (podcasts + posts) that have a reference to the selected tags.
When the app loads, all tags are selected ->
When the user checks or unchecks a tag -> re-query

All of this should be handled by grok query strings, very similar to graphql
https://www.sanity.io/docs/query-cheat-sheet

from the cheatsheet I see this query might work:
```ts
// documents that reference tags with slugs of "tag1" or "tag2"
*[count((tags[]->slug.current)[@ in ["tag1", "tag2"]]) > 0]
```

**Comment by Atmosfearful** [comment]
@ladytrekker 
# Checkboxes 
All sounds good to me.

> Do we need a "Reset" icon here to reset all filters?

I say no, page refresh is fine

# Text Search

After thinking about this, I believe 👉 *text queries should always search across all tags*. So ignoring what the user has checked. I posted in discord about to ask for more opinions.

One more detail: when you clear the text search input and press Enter (or tap the Search icon) it should re-query *with whatever tags they had selected before*. So we can not just fall back to `props.documents`, we have to re-query

A reset icon is nice but not necessary for v1 but if you have time, go for it. If not, please create a ticket for follow up. I imagine this:
- If text is present, show an X next to the search icon (we still need the search icon button for mobile users).
- When you click the X, it clears the input and re-queries.

After that, here is a further enhancement idea for later (low priority) https://github.com/KlimaDAO/klimadao/issues/679

# Sort by Select

Good details about Relevance, I didn't think about that extra complexity.
- I agree, text search === force sort by relevance
- Yes it is nice if the user can do text search first by relevance -> and then sort by date. This is not a *must have* feature but it should be easy ya?
- If the user clears the text search, we go back to "newest first" regardless of their previous sorting
- If they create a new text search, we go back to "relevance" regardless of their previous sorting

so all together we fire a new query whenever:
1. `search input` is changed
2. `tag` is checked (and `search input` is empty)
3. `tag` is unchecked (and `search input` is empty)
4. `sort` is changed


# No search result

👉 If all checkboxes are unchecked, we should query as if *all are checked* (default query) because the user would never want search for nothing.

This means the only time we ever see "no results found" is when they are typing in the text search input.

**Comment by rittaaka** [comment]
@jabby09 This is my Github account, feel free to tag me on items whenever they're my to-do's. I'll try to get proactive once I know where to look.

**Comment by Atmosfearful** [comment]
I wrote out this big beautiful post with nicely formatted pseudocode queries then accidentally closed the tab and lost it all 😡 😡 😡 😡 

I think we disagree on additive vs narrowing.

You wrote:

> If all checkboxes all selected by default then this would imply: the sum of all visible documents need to have all tags attached

I disagree. If I go on newegg and check all manufacturers + all prices, I get a query like this:
- `includesOneOf(manufacturer1, manufacturer2) && includesOneOf(price1, price2)`

I think tags are the same. More checked boxes -> more matches

Whether we default all-checked or none-checked doesn't matter actually, just a design decision. But we definitely want to use a set of "OR" matches,

like `tag1 OR tag2 OR tag3`.

Some filters like airbnb use `AND`, give me a hotel with a `pool && wifi && queenBed`. This is not what we want

In pseudocode the grok query is more like this:
```ts
const mediums = ["post", "podcast"];
const tags = ["basics", "policy"];
const queryMatch = mediums.includesOneOf(document._type) && document.tags.includesOneOf(tags);
```

Just need to express this in Grok and handle the edge cases/fallbacks.

I don't think this lines up exactly with what you describe in your points but I hope it clarifies. Please let me know if I missed something.

**Comment by ladytrekker** [comment]
> 
> In pseudocode the grok query is more like this:
> 
> ```ts
> const mediums = ["post", "podcast"]; // if empty, fallback to ALL_MEDIUMS
> const tags = ["basics", "policy"]; // if empty, fallback to ALL_DEFAULT_TAGS
> const queryMatch = mediums.includesOneOf(document._type) && document.tags.includesOneOf(tags);
> ```
> 
> Just need to express this in Grok and handle the edge cases/fallbacks.

If I get you right, this is how it's already working.
So we are talking about the same: More checked boxes -> more matches.

And if you check only "podcast" and "user guide", then there is no match because no podcast has this tag.

I pushed a commit now with the checkboxes being enabled by default.
To me this is seems more confusing. Especially after a text search.
I would go for unchecked by default 🦐 

You can play around with the Preview Link to check the behavior (watch the network tab for the documents tags data). 

Here is the PR again => https://github.com/KlimaDAO/klimadao/pull/689

**Comment by jabby09** [comment]
@ShimonD-KlimaDAO @rittaaka 

The new resource page is ready for QA in staging (https://staging-site.klimadao.finance/resources). I will be testing as well.

There is a content issue related to the images in the featured carousel in that they have overlapping text which will need to be fixed.

As a reminder...this is the f[igma design](https://www.figma.com/file/ITpEVoi0pVYDRSLpFOaCiz/LANDINGPAGE?node-id=1747%3A1763)

**Comment by ShimonD-KlimaDAO** [comment]
Waiting to get access to the figma before testing. Should be able to provide feedback by tomorrow.

**Comment by ShimonD-KlimaDAO** [comment]
Feedback: #774

**Comment by jabby09** [comment]
@ladytrekker should we close this issue now?

**Comment by ladytrekker** [comment]
@jabby09 Yes, true.
We can open a new one for the improvement ideas as soon as these can be formulated
